db/postgres/bookings.go

```diff
@@ -147,6 +147,39 @@ func (p PostgresDBStore) CreateBooking(booking *model.Booking) (string, error) {
 		return "", errors.New("invalid operation: workspace is not offered")
 	}
 
+	// Check if user has a booking at this time
+	err = tx.QueryRow(
+		`SELECT count(*) FROM bookings 
+					WHERE user_id=$1 AND cancelled=FALSE AND
+                    	   (start_time <= $2 AND end_time >= $2) OR
+                    	   (start_time <= $3 AND end_time >= $3) OR
+                    	   (start_time >= $2 AND end_time <= $3)
+                    	   `,
+		// todo: pray can you proof check that ORs 1 and 2 handle this case (start_time <= $2 AND end_time >= $3)
+		booking.UserID, booking.StartDate, booking.EndDate,
+	).Scan(&count) // if sql query fails, count won't be "updated" -> check error
+	if err != nil || count != 0 {
+		return "", errors.New("invalid operation: the user has a booking in this date range")
+	}
+
+	// Max 10 Bookings per user
+	loc, err := time.LoadLocation("America/Vancouver")
+	if err != nil {
+		return "", errors.New("invalid operation: location failed")
+	}
+	timeNow := time.Now().In(loc)
```

Comment:
i dont think we have to do this, plus even if u do; u should do it in UTC time rather than PST